### PR TITLE
Dev network retrofit

### PR DESCRIPTION
**Pull request by sergei-kabak-idf**
retrofit and gen code

### CODE REVIEW

File: packages/data/pubspec.yaml [hunk 1]
```diff
@@ -10,50 +10,55 @@ environment:
...
-  #         style: italic
-  #   - family: Trajan Pro
-  #     fonts:
-  #       - asset: fonts/TrajanPro.ttf
-  #       - asset: fonts/TrajanPro_Bold.ttf
-  #         weight: 700
-  #
-  # For details regarding fonts in packages, see
-  # https://flutter.dev/custom-fonts/#from-packages
+# To add assets to your package, add an assets section, like this:
```
Comment (alexei-kruk-idf) [inline packages/data/pubspec.yaml]: можно убрать комменты

File: pubspec.yaml [hunk 2]
```diff
@@ -34,7 +34,6 @@ dependencies:
 
 
 
-
   # The following adds the Cupertino Icons font to your application.
```
Comment (alexei-kruk-idf) [inline pubspec.yaml]: Тоже комменты)

**Review by alexei-kruk-idf — APPROVED** [review]

**Review by yuliya-klescheva-idf — APPROVED** [review]

File: packages/data/.gitignore [hunk 3]
```diff
@@ -30,3 +30,4 @@ CHANGELOG.md
 README.md
 LICENSE
 build/
+/lib/service/retrofit_api_service.g.dart
```
Comment (sergei-mikhailovskii-idf) [inline packages/data/.gitignore]: avoid specifying such concrete paths. better variant - ignore file retrofit_api_service.g.dart anywhere. the best - ignore  all .g.dart files anywhere

File: packages/data/lib/dio/dio_builder.dart [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import 'package:injectable/injectable.dart';
+
+@module
+abstract class DioBuilder {
+  Dio dioBuilder() {
+    final baseOptional = BaseOptions(baseUrl: baseUrl);
+    final client = Dio(baseOptional);
+    client.interceptors.add(
+      LogInterceptor(
+          request: false, responseHeader: false, requestHeader: false,),
```
Comment (sergei-mikhailovskii-idf) [inline packages/data/lib/dio/dio_builder.dart]: formatting - press command + option + L if u r working in studio. in vscode - idk

File: packages/data/lib/mapper/mapper.dart [hunk 5]
```diff
@@ -0,0 +1,5 @@
+abstract class Mapper {
```
Comment (sergei-mikhailovskii-idf) [inline packages/data/lib/mapper/mapper.dart]: good idea, but not the best implementation - u should create `abstract class Mapper<I, O>`. I - input type, O - output type. this class should have method call - `O call(I params)`. then create subclass, `IsPalindromeMapper implements Mapper<Map<String, dynamic>, bool>`, override method call. and then create the mapper object and use it anywhere u need

File: packages/data/lib/service/retrofit_api_service.dart [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import 'package:dio/dio.dart';
```
Comment (sergei-mikhailovskii-idf) [inline packages/data/lib/service/retrofit_api_service.dart:1]: formatting

File: packages/domain/lib/use_case/palindrome_use_case.dart [hunk 7]
```diff
@@ -1,19 +1,19 @@
...
 
+part 'use_case.dart';
 
-class PalindromeUseCaseImpl implements UseCase {
-  PalindromeUseCaseImpl(this._palindromeRepositoryImpl);
+class PalindromeUseCaseImpl implements UseCase<String, Future<bool>> {
+  PalindromeUseCaseImpl(this._repository);
 
-  final PalindromeRepositoryImpl _palindromeRepositoryImpl;
+  final PalindromeRepositoryImpl _repository;
```
Comment (sergei-mikhailovskii-idf) [inline packages/domain/lib/use_case/palindrome_use_case.dart]: u should work with PalindromeRepository (abstraction) here, but not implementation

**Review by sergei-mikhailovskii-idf — CHANGES_REQUESTED** [review]

File: packages/domain/pubspec.yaml [hunk 8]
```diff
@@ -13,6 +13,7 @@ dependencies:
   meta: ^1.7.0
   get_it: ^7.2.0
   injectable: ^1.5.3
+  json_serializable: ^6.1.5
 
   data:
```
Comment (sergei-mikhailovskii-idf) [inline packages/domain/pubspec.yaml]: very important point - domain should not know anything about data

**Review by sergei-mikhailovskii-idf — CHANGES_REQUESTED** [review]

File: packages/domain/pubspec.yaml (hunk 8 above)
Comment (sergei-kabak-idf) [reply packages/domain/pubspec.yaml]: fixed

File: packages/data/pubspec.yaml (hunk 1 above)
Comment (sergei-kabak-idf) [reply packages/data/pubspec.yaml]: fixed

File: pubspec.yaml (hunk 2 above)
Comment (sergei-kabak-idf) [reply pubspec.yaml]: fixed

File: packages/data/.gitignore (hunk 3 above)
Comment (sergei-kabak-idf) [reply packages/data/.gitignore]: fixed

File: packages/data/lib/dio/dio_builder.dart (hunk 4 above)
Comment (sergei-kabak-idf) [reply packages/data/lib/dio/dio_builder.dart]: fixed

File: packages/data/lib/mapper/mapper.dart (hunk 5 above)
Comment (sergei-kabak-idf) [reply packages/data/lib/mapper/mapper.dart]: delete class

File: packages/data/lib/service/retrofit_api_service.dart (hunk 6 above)
Comment (sergei-kabak-idf) [reply packages/data/lib/service/retrofit_api_service.dart:1]: fixed

File: packages/domain/lib/use_case/palindrome_use_case.dart (hunk 7 above)
Comment (sergei-kabak-idf) [reply packages/domain/lib/use_case/palindrome_use_case.dart]: fixed

**Pull request merged**